public/app/features/datasources/components/BasicSettings.tsx

```diff
@@ -1,18 +1,43 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 
 import { selectors } from '@grafana/e2e-selectors';
-import { InlineField, InlineSwitch, Input } from '@grafana/ui';
+import { getDataSourceSrv } from '@grafana/runtime';
+import { InlineField, InlineSwitch, Input, Alert } from '@grafana/ui';
+
+const useInitHasAlertingEnabled = (dataSourceName: string) => {
+  const [hasAlertingEnabled, setHasAlertingEnabled] = useState(false);
+
+  useEffect(() => {
+    const initDataSourceAlertingEnabled = async () => {
```

Comment:
We should be able to pass `dataSourceMeta` from `<EditDataSource />` so we don't have to do another api call just to fetch the meta again.

Reply:
Actually, there is no additional api call as this is cached by rxjs

Reply:
We can try to use `getInstanceSettings` instead of `get`. In some cases `get` may require an additional API call (there is a condition that sometimes may cause a request, hence it returns a promise instead of a plain object), whereas `getInstanceSettings` returns an object instead of a promise, and doesn't do any API calls

Reply:
This way we could also get rid of the `async` logic here

Reply:
done!